temporalio/worker/_workflow_instance.py

```diff
@@ -642,11 +640,13 @@ async def run_query() -> None:
                         raise ValueError(
                             f"Expected 1 result payload, got {len(result_payloads)}"
                         )
-                    command.respond_to_query.succeeded.response.CopyFrom(
-                        result_payloads[0]
-                    )
+                command = self._add_command()
+                command.respond_to_query.query_id = job.query_id
+                command.respond_to_query.succeeded.response.CopyFrom(result_payloads[0])
             except Exception as err:
                 try:
+                    command = self._add_command()
+                    command.respond_to_query.query_id = job.query_id
```

Comment:
Non-blocking: I assume these lines cannot raise an exception; I would put them outside the `try`.

Reply:
They were outside of try in the previous code, but the purpose of this fix is to not invoke these until it's time to make the command because if you call this _before_ an exception is raised, you'll have an unfinished command that was added